logos_data_loader.py

```diff
@@ -0,0 +1,144 @@
+# Data loader for Logos data, load data and transform data to correct format if needed
+# For binary relation extraction
+
+import json 
+from itertools import combinations
+from typing import List
+
+class LogosDataLoader:
+    """Data loader for Logos data
+
+    Load data and transform data to correct format if needed
+    For binary relation extraction
+
+    Attributes:
+        data_path: Path to file contain data
+        max_data_size: Maximum number of data points to use
+        is_PURE_format: If given file contains data in PURE format 
+        data: Data in correct format
+    """
+
+    def __init__(self, data_path: str, max_data_size: int = 50000, is_PURE_format: bool = False) -> None:
+        """Initialize LogosDataLoader 
+
+        Args:
+            data_path: Path to file contain data
+            max_data_size: Maximum number of data points to use
+            is_PURE_format: If given file contains data in PURE format 
+        """
+
+        self.data_path = data_path
+        self.max_data_size = max_data_size
+        self.is_PURE_format = is_PURE_format
+        self.data = None
+
+    def run(self) -> List[dict]:
+        """Run LogosDataLoader 
+
+        Load data, and transform if specified so
+        Data length is at most max_data_size
+
+        Returns:
+            List[dict]: data in correct format
```

Comment:
nit: You can write 'data in correct format' only

You don't need to put the type here because the function signatures shows it